transmart-core-db/src/main/groovy/org/transmartproject/db/multidimquery/query/Constraint.groovy

```diff
@@ -205,6 +207,7 @@ class BiomarkerConstraint extends Constraint {
  */
 @Canonical
 class ModifierConstraint extends Constraint {
+    static String constraintName = "modifier"
```

Comment:
This convention already does not hold for all constraint classes (Negation, Combination). Also I'm not sure how to implement that since these are static properties. The `static String getName()` would need to have access to the (derived) Class object somehow. The constraints could be refactored from statics to be instances like we do with DimensionImpl subclasses but that is a bigger change. The entire constraint building & querying should be made more consistent and use the Hibernate 5 features, but that's to big to do now. I wanted to do the minimal changes to get the public api in good shape, internals can be refactored later (if there is budget).